### PR TITLE
Components: Modal, add story

### PR DESCRIPTION
**Pull request by ItsJonQ**
## Components: Modal, Add story

Screen Capture on 2019-10-23 at 13-37-27

This update adds a story for the Modal component.
The story showcases an updated and enhanced example from the Modal
README. It also includes a series of knobs to better demonstrate the
Modal props.

Screen Shot 2019-10-23 at 1 37 37 PM

### CODE REVIEW

File: packages/components/src/modal/stories/index.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import Modal from '../';
+
+/**
+ * WordPress dependencies
+ */
+import { Fragment, useState } from '@wordpress/element';
+
+export default { title: 'Modal', component: Modal };
+
+export const _default = () => {
```
Comment (gziolo) [inline packages/components/src/modal/stories/index.js]: This one still needs to be refactored to move hooks out of the story as discussed in other PRs :)

Comment (ItsJonQ) [reply packages/components/src/modal/stories/index.js]: @gziolo Ah yes! Thank you. Will get on that :). I think this was an earlier Storybook submission I did, before I discovered the `<Example />` technique.

**Review by gziolo — APPROVED** [review]
As noted in my comment, this modal opens by default:
Screen Shot 2019-10-30 at 15 47 25

I see one accessibility-related issue. It might be a bug in the current implementation. We should double-check and file an issue if my guess is true.

In general, this work is great. My comments are nits 😍

File: packages/components/src/modal/stories/index.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+/**
+ * WordPress dependencies
+ */
+import { Fragment, useState } from '@wordpress/element';
+
+export default { title: 'Modal', component: Modal };
+
+const ModalExample = ( props ) => {
+	const [ isOpen, setOpen ] = useState( true );
```
[the remark this answers is not in the document]
Comment (ItsJonQ) [reply packages/components/src/modal/stories/index.js]: @gziolo Originally, I was going to add the state via a Storybook knob. The downside is... you can't update a knob's state. It's one way.

We would need this if we manually closed the `Modal` by clicking `X`.

[the remark this answers is not in the document]
Comment (gziolo) [reply packages/components/src/modal/stories/index.js]: Yes, it doesn't work as I would expect. @brentswisher raised the same issue on other PR. I tested it myself and we should ignore this comment 😅

**Comment by gziolo** [comment]
> @gziolo Refactored the `Modal` story so that it doesn't open by default. It looks like that resolves the a11y issue as well 💪

It only hides them as far as I can tell :) The issue exists. I updated `axe-core` together with `puppeteer` upgrade and it reports issues with modal in many places. See:
https://github.com/WordPress/gutenberg/pull/18205#issuecomment-548334642

Anyway, it's not an issue with this PR.

File: packages/components/src/modal/stories/index.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+	const openModal = () => setOpen( true );
+	const closeModal = () => setOpen( false );
+
+	return (
+		<Fragment>
```
[the remark this answers is not in the document]
Comment (gziolo) [reply packages/components/src/modal/stories/index.js]: That's the best part, you don't have to import `createElement` and `Fragment` when using WordPress preset for Babel 😍

[the remark this answers is not in the document]
Comment (gziolo) [reply packages/components/src/modal/stories/index.js]: I updated the example in docs as well.

**Pull request merged**